test.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+#
+# This is the InfluxDB test script.
+# This script can run tests in different environments.
+#
+# Usage: ./test.sh <environment_index>
+# Corresponding environments for environment_index:
+#      0: normal 64bit tests
+#      1: tsm 64bit tests
+#      2: race enabled 64bit tests
+#      3: normal 32bit tests
+#      *: to run all tests in parallel containers
+
+# Get dir of script and make it is our working directory.
+DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
+cd $DIR
+
+ENVIRONMENT_INDEX=$1
+# Set default parallelism
+PARALLELISM=${PARALLELISM-1}
+# Set default timeout
+TIMEOUT=${TIMEOUT-480s}
+
+# Default return code 0
+rc=0
+
+if [ -z "$OUTPUT_DIR" ]
+then
+    echo "Please specify an OUTPUT_DIR env var for test logs"
+    exit 1
+fi
+
+
+
+# Executes the given statement, and exits if the command returns a non-zero code.
+function exit_if_fail {
+    command=$@
+    echo "Executing '$command'"
+    $command
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "'$command' returned $rc."
+        exit $rc
+    fi
+}
+
+# Run a test in a docker container
+# Usage: run_test_docker <Dockerfile> <test_name>
+function run_test_docker {
+    local dockerfile=$1
+    shift
+    local name=$1
+    shift
+    local dockerimage="$HOME/docker/${dockerfile}.tar"
+    local logfile="$OUTPUT_DIR/${name}.log"
+
+    echo "Running test in docker $name with args $@"
+
+    if [[ -e "$dockerimage" ]]; then docker load -i "$dockerimage"; fi
+    exit_if_fail docker build -f "$dockerfile" -t "$name" .
+    mkdir -p ~/docker; docker save "$name" > "$dockerimage"
+    docker run \
+         --rm \
+         -v "$DIR:/root/go/src/github.com/influxdb/influxdb" \
```

Comment:
@otoolep Right here the docker command mounts the current dir (which is the code) into the container. The container only has the dependencies necessary to build and test the code but not the code itself. It expects the code to be mounted in when started.